README.md

```diff
@@ -404,15 +404,97 @@ Note that named function signatures in an interface block indicate methods, rath
 
 ```js
 interface Collection {
-  (signatureParam: Any) => Any, // Collection() signature
-  method1(items: [...Array]) => [...Array], // method
+  (signatureParam: Any) => Any,              // Collection() signature
+  method1(items: [...Array]) => [...Array],  // method
   method2(items: [...Object]) => [...Object] // method
 }
+```
+
+### Key Definition
+
+For convenience, interfaces that feature similar keys can use property templates.
+These templates let us define labels—and optionally types—of objects' keys.  
+*Internally* there are only 2 valid types of keys: `String` and `Symbol`.
+In this regard, [regular expression literal](#regular-expression) and [`Number`](#number) will be automatically mapped to `String`. If ommited the type defaults to `String`.
+
+```js
+{
+  [id1]: {
+    skating: {time: 1000, money: 300},
+    'cooking': {time: 9999, money: 999}
+  },
+  [id2]: {
+    "jogging": {time: 300, money: 0}
+  }
+  etc.
+}
+```
+
+The preceding object can be expressed using this interface:
+
+```js
+interface clientHobbies {
+  [id: Symbol]: {
+    [hobby]: {          // [hobby: String]: {
+      time: Number,
+      money: Number
+    }
+  }
+}
+```
+
+You can even go further to improve the semantics of your interfaces:
+
+```js
+interface Expenditure {
+  time: Number,
+  money: Number
+}
+
+interface clientHobbies {
+  [id: Symbol]: {
+    [hobby]: Expenditure
+  }
+}
+```
+
+#### Number
+
+Types that are automatically coerced into a string can also be documented:
```

Comment:
We are probably not talking about the same thing. `String(Number)`—which is not in the current version—would roughly mean that it accepts stringified numbers. Something that is closer to the actual internal behaviour but that wouldn't add much for the user.